tools/crm_mon.c

```diff
@@ -1097,6 +1097,24 @@ detect_user_input(GIOChannel *channel, GIOCondition condition, gpointer user_dat
 {
     int c;
     gboolean config_mode = FALSE;
+    gboolean rc = TRUE;
+
+    /* If the attached pty device (pseudo-terminal) has been closed/deleted,
+     * the condition (G_IO_IN | G_IO_ERR | G_IO_HUP) occurs.
+     * Exit with an error, otherwise the process would persist in the
+     * background and significantly raise the CPU usage.
+     */
+    if ((condition & G_IO_ERR) && (condition & G_IO_HUP)) {
+        rc = FALSE;
+        clean_up(CRM_EX_IOERR);
+    }
+
+    /* The connection/fd has been closed. Refresh the screen and remove this
+     * event source hence ignore stdin.
+     */
+    if (condition & (G_IO_HUP | G_IO_NVAL)) {
+        rc = FALSE;
+    }
```

Comment:
Just to be safe, let's return immediately after this if G_IO_IN is not set

Also, this would be a good time to use G_SOURCE_CONTINUE and G_SOURCE_REMOVE instead of TRUE and FALSE

Reply:
> Just to be safe, let's return immediately after this if G_IO_IN is not set

Makes sense.

> 
> Also, this would be a good time to use G_SOURCE_CONTINUE and G_SOURCE_REMOVE instead of TRUE and FALSE

I was actually checking these:
https://docs.gtk.org/glib/callback.SourceFunc.html#return-value
https://docs.gtk.org/glib/callback.IOFunc.html#return-value

, but didn't find G_SOURCE_* mentioned in the latter.

I guess it's fine and beneficial to use them for GIOFunc anyway.